core/base/bottleneckDistance/BottleneckDistanceMainImpl.h

```diff
@@ -102,7 +102,7 @@ int BottleneckDistance::computeBottleneck(const std::vector<diagramTuple> &d1,
         const diagramTuple a, const diagramTuple b) -> dataType {
     BNodeType ta1 = std::get<1>(a);
     BNodeType ta2 = std::get<3>(a);
-    double w = wasserstein > 1 ? wasserstein : 1; // L_inf not managed.
+    const int w = wasserstein > 1 ? wasserstein : 1; // L_inf not managed.
```

Comment:
is w the exponent used in the expression of the distance?
if so, in principle, this can be a real value (not necessarily an integer).

would that be possible that powInt behaves differently based on the type it is given as an argument?
for instance, if it detects the argument is indeed an integer, use the current powInt implementation.
if it detects it is not, use the regular pow implementation.

Reply:
Here, `w` depends on the value of the `wasserstein` variable, which is an integer defined at line 41 of the same file. In this case, using a double type should be less efficient.

I tried to keep using `std::pow` every time the exponent was clearly a double variable.

In theory, it should be possible to unite the two behaviors using `std::is_integral` and `std::is_floating_point`. I will check that.